apps/web/composables/useModernImage/useModernImage.ts

```diff
@@ -32,8 +34,50 @@ export const useModernImage: UseModernImageReturn = () => {
     return images.map((image: AgnosticImage) => ({ ...image, url: addModernImageExtension(image.url) }));
   };
 
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  const getImageForViewport = (product: Product, context: string): string => {
+    if (!product) return '';
+
+    const { isDesktop, isTablet } = useBreakpoints();
+    const isLargeViewport = isDesktop || isTablet;
+
+    switch (context) {
+      case 'ItemList': {
+        return productGetters.getPreviewImage(product);
+      }
+      case 'Whislist': {
+        return isLargeViewport ? productGetters.getFullImage(product) : productGetters.getMiddleImage(product);
+      }
+      case 'CartProductCard': {
+        return isLargeViewport
+          ? productGetters.getSecondPreviewImage(product)
+          : productGetters.getPreviewImage(product);
+      }
+      default: {
+        return '';
+      }
+    }
+  };
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  const addWebpExtensionForGallerySfImages = (images: ImagesData[]) => {
```

Comment:
```suggestion
  const addModernImageExtensionForSfImages = (images: SfImage[]) => {
```
should be using existing function